.github/workflows/coverage.yml

```
@@ -0,0 +1,46 @@
+name: Code Coverage Workflow
+
+on:
+  schedule:
+    - cron: '0 0 * * 1'
+  pull_request:
```

```suggestion
  push:
    branches:
      - main
      - release-*
```

Suggest not running this on PRs. Manual runs can still be triggers in the actions page thanks to `workflow_dispatch`.

---

Ok. I'll make this change before we merge... For now we need it to run on this PR to test it.